### PR TITLE
SiluAndMulWithClamp rework for vLLM IR

### PR DESCRIPTION
**Pull request by bohnstingl**
## Purpose

This PR rewrites the `forward_*` methods in `SiluAndMulWithClamp` with a call to new `vllm.ir.ops.silu_and_mul_with_clamp`. This is part to resolve the issue https://github.com/vllm-project/vllm/issues/40406

## Test Plan

Specific tests for the new vLLM IR have been added. In addition, I checked correctness and latency for the `deepseek-ai/DeepSeek-V4-Flash` model.

## Test Result

I ran `lm_eval` tests on 4xH100s

```
vllm serve --model deepseek-ai/DeepSeek-V4-Flash --tensor-parallel-size 4 --quantization fp8 --max-model-len 4096 --gpu-memory-utilization 0.7 --enforce-eager --kv-cache-dtype fp8 --block-size 256 --enable-expert-parallel --tokenizer-mode deepseek_v4

lm_eval --model local-completions --model_args pretrained=deepseek-ai/DeepSeek-V4-Flash,base_url=http://0.0.0.0:8000/v1/completions,num_concurrent=50,max_retries=3 --tasks gsm8k --num_fewshot 5 --batch_size auto
```

**main:**                                                                                                    
                                                                                                               
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9560|±  |0.0056|
|     |       |strict-match    |     5|exact_match|↑  |0.9568|±  |0.0056|           
                                                                                                               
  **branch:**                                                                                           
                                                                                                               
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9522|±  |0.0059|
|     |       |strict-match    |     5|exact_match|↑  |0.9530|±  |0.0058|


In addition, I ran some latency experiments on 4xH100s:
```
vllm bench latency --model deepseek-ai/DeepSeek-V4-Flash --tensor-parallel-size 4 --quantization fp8 --max-model-len 4096 --gpu-memory-utilization 0.7 --enforce-eager --kv-cache-dtype fp8 --block-size 256 --enable-expert-parallel --tokenizer-mode deepseek_v4
```

| Metric | Latency/s (main) | Stderr (main) | Latency/s (branch) | Stderr (branch) |
|--------|------------------:|--------------:|--------------------:|----------------:|
| Avg    |         13.341037 |      0.144069 |           13.915080 |        0.239610 |
| p10    |         13.203982 |      0.149102 |           13.690670 |        0.414609 |
| p25    |         13.278748 |      0.165272 |           13.790511 |        0.379956 |
| p50    |         13.343506 |      0.157119 |           13.946527 |        0.177370 |
| p75    |         13.401830 |      0.126252 |           14.037564 |        0.152943 |
| p90    |         13.441323 |      0.112400 |           14.098922 |        0.126524 |
| p99    |         13.658659 |      0.150968 |           14.156002 |        0.144399 |

**NOTE:** The latency benchmark for this model was a bit noisy, but I don't think there is any substantial difference between the vLLM IR and the original implementation.


---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [X] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [X] The test plan, such as providing test command.
- [X] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: tests/kernels/ir/test_activation.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        torch.testing.assert_close(
+            out_up,
+            torch.tensor(
+                [[silu_1 * limit]],
+                dtype=torch.float32,
+                device=current_platform.device_type,
+            ),
+            atol=1e-3,
+            rtol=1e-3,
+        )
```
[the remark this answers is not in the document]
Comment (bohnstingl) [reply tests/kernels/ir/test_activation.py:109]: I updated the tests to use the `dtype` parameter and not the fixed `torch.float32`

File: vllm/model_executor/layers/activation.py [hunk 2]
```diff
@@ -169,26 +172,14 @@ class SiluAndMulWithClamp(CustomOp):
...
     def forward_cuda(self, x: torch.Tensor) -> torch.Tensor:
-        d = x.shape[-1] // 2
-        output_shape = x.shape[:-1] + (d,)
-        out = torch.empty(output_shape, dtype=x.dtype, device=x.device)
-        self.op(out, x, self.swiglu_limit)
-        return out
-
-    def forward_xpu(self, x: torch.Tensor) -> torch.Tensor:
-        return self.forward_cuda(x)
+        return ir.ops.silu_and_mul_with_clamp(x, self.swiglu_limit)
```
[the remark this answers is not in the document]
Comment (bohnstingl) [reply vllm/model_executor/layers/activation.py:182]: I re-added the `forward_xpu` function.
